waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/util/SaslHelper.java

```diff
@@ -42,7 +43,22 @@
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
 public final class SaslHelper {
 
-  public static HadoopThriftAuthBridge.Server createSaslServer(HiveConf conf) throws TTransportException {
+  public static class SaslServerAndMDT {
```

Comment:
remove this doesn't belong in a helper class, this solution needs some more thought and refactoring.

I suggest to look at how to refactor the delegationTokenManager out and see how we can inject that using beans. Your relying too much on static methods and classes. I do not have more time to go into more detail but please have a look and see if you can refactor this.